VerilogCodingStyle.md

```diff
@@ -2610,6 +2611,79 @@ module mymod (
 endmodule
 ```
 
+#### Continuous Assignment from Functions
+
+Continuous assignment from a function return value is prohibited when the
+function contains interior sensitivities.
+
+According to the LRM, `assign` statements only consider changes in the RHS
+expressions but not those inside functions. As a result, the `assign`
+statement might not update as intended in simulation, while it might behave
+correctly in synthesis.
+
+To ensure predictable and consistent behavior across different tools, it is
+advisable to use the `always_comb` construct instead of `assign` when
+continuous assignments involve function calls with interior sensitivities. The
+`always_comb` block automatically evaluates whenever any signal or variable
+within the expression changes, including those inside functions, thereby
+guaranteeing proper updates and reliable code behavior.
+
+Note that `always @*` is also only sensitive to changes to the arguments of a
+function. Therefore, it suffers from the same limitation as `assign` and should
+also be avoided for functions that contain interior sensitivities.
+
+&#x1f44e;
+```systemverilog {.bad}
+module mymod (
+  input        in_i,
+  output logic out_o
+);
+
+function automatic myfunc;
+  return in_i;
+endfunction
+
+assign out_o = myfunc(); // bad
+
+endmodule
+```
+
+&#x1f44d;
+```systemverilog {.good}
+module mymod (
+  input        in_i,
+  output logic out_o
+);
+
+function automatic myfunc;
+  return in_i;
```

Comment:
I think the fact that functions reference anything non-local is bad.

The whole "continuous assignment from function" issue is moot if we just disallow the function from referencing anything non-local.